### PR TITLE
Add support for upgrading clips in a Timeline

### PR DESCRIPTION
**Pull request by amechtley**
Similar to Animator, we check for PlayableDirector with references to PlayableAsset, and check that PlayableAsset for sub-assets.

Since we can't know what the binding key actually is (without assuming it is a Timeline, and thus adding a dependency on Timeline), I just iterate through all animator binding values and check their usage. It is technically possible a Timeline could have bindings for 2 different animators, which may differ in their usage (e.g., one uses an upgradable material and the other not; one has a path to a renderer and the other does not), in which case we couldn't upgrade. But I figured this would be better than catching zero Timeline use cases, and would likely catch many or most of them.

### CODE REVIEW

**Comment by amechtley** [comment]
FYI @simonbz, as with the base PR, most of this is just resources for the test project. The interesting code is just in AnimationClipUpgrader.cs

**Comment by simonbz** [comment]
Pushed some more modifications to support standalone clips referenced in timeline assets.
@amechtley Can you have a look before I merge that back into master PR.

**Review by simonbz — APPROVED** [review]

**Review by amechtley — COMMENTED** [review]
Only one question, otherwise lgtm

File: com.unity.render-pipelines.universal/Editor/AnimationClipUpgrader.cs [hunk 1]
```diff
@@ -278,13 +283,53 @@ IEnumerable<ClipPath> clipPaths
...
+                // get all clip sub-assets
+                var clips = new HashSet<IAnimationClip>(
+                    AssetDatabase.LoadAllAssetsAtPath(assetPath)
+                        .Where(asset => asset is AnimationClip)
+                        .Select(asset => (IAnimationClip)(AnimationClipProxy)(asset as AnimationClip))
+                );
+
+                // get all clip dependency-assets
+                clips.UnionWith(AssetDatabase.GetDependencies(assetPath)
+                    .Select(dependencyPath => AssetDatabase.LoadAssetAtPath<AnimationClip>(dependencyPath))
```
[the remark this answers is not in the document]
Comment (simonbz) [reply com.unity.render-pipelines.universal/Editor/AnimationClipUpgrader.cs:305]: Was wondering about that. A FBX could have a bunch of clips, but the timeline may only refer to a single one of them. How would we get that single clip out of the FBX file? Is there a way to do that?

[the remark this answers is not in the document]
Comment (amechtley) [reply com.unity.render-pipelines.universal/Editor/AnimationClipUpgrader.cs:305]: I don't think we have a straightforward way to do it. We may be safe here inasmuch the common use case of multiple clips on a single asset (FBX) is also the case least likely to have curves for material properties. I suggest we just document it as a limitation in a comment here.

(btw I think you can also just pass the method group directly `.Select(AssetDatabase.LoadAssetAtPath<AnimationClip>)`)

[the remark this answers is not in the document]
Comment (simonbz) [reply com.unity.render-pipelines.universal/Editor/AnimationClipUpgrader.cs]: done in 8791966048533aef3e65b37aa537473e62a28fb4

**Pull request merged**